pkg/gardener/shoot/extender/provider/provider.go

```diff
@@ -37,18 +38,13 @@ func NewProviderExtenderForCreateOperation(enableIMDSv2 bool, defMachineImgName,
 			return err
 		}
 
-		infraConfig, controlPlaneConf, err := getConfig(rt.Spec.Shoot, workerZones)
+		infraConfig, controlPlaneConf, err := getConfig(rt.Spec.Shoot, workerZones, nil)
 		if err != nil {
 			return err
 		}
 
 		controlPlaneConf, infraConfig = overrideConfigIfProvided(rt, infraConfig, controlPlaneConf)
```

Comment:
we should stop ovveriding Config with values from RuntimeCR as we are making deprecated fields `spec.shoot.provider.infrastructureConfig` and `spec.shoot.provider.controlPlaneConfig`

Reply:
Done